site/en/blog/show-picker/index.md

```diff
@@ -0,0 +1,213 @@
+---
+layout: 'layouts/blog-post.njk'
+title: 'Show a browser picker for date, time, color, and files'
+description: >
+  The web platform now ships with a canonical way to show a browser picker.
+authors:
+  - beaufortfrancois
+date: 2022-01-11
+hero: image/vvhSqZboQoZZN9wBvoXq72wzGAf1/zWT5kk3Q5RlDNqGkdgis.jpeg
+alt: A photo of a monthly schedule printed on white paper
+tags:
+  - chrome-99
+---
+
+For a long time, you had to resort to custom widget libraries or hacks to show a
+date picker. The web platform now ships with `HTMLInputElement.showPicker()`, a
+canonical way to show a browser picker not only for dates, but also time, color,
+and files.
+
+## Background {: #background }
+
+A [frequent request] we hear from web developers is:
+
+<blockquote>
+  <p>
+    How do I programmatically<br/>
+    show a picker for controls like date?
+  </p>
+  <cite>
+    Stack Overflow
+  </cite>
+</blockquote>
+
+Current answers are not great; they rely on external libraries, CSS hacks, or
+specific browser behaviors like simulating a user interaction with `click()`.
+
+I'm happy to share that those days will be over soon as the web platform is
+introducing a canonical way to show a browser picker for `<input>` elements with
+these types: `"date"`, `"month"`, `"week"`, `"time"`, `"datetime-local"`,
+`"color"`, and `"file"`. It will also work for `<input>` elements with
+suggestions powered by `<datalist>` or `"autocomplete"` which we'll cover as
+well in this article.
+
+<figure class="w-figure">
+{% Img src="image/vvhSqZboQoZZN9wBvoXq72wzGAf1/uh0U2YQnUMato21MzzbR.png", alt="Screenshots of browser pickers", width="800", height="217" %}
+  <figcaption class="w-figcaption">Browser date pickers in Chrome desktop, Chrome mobile, Safari desktop, Safari mobile, and Firefox desktop (July 2021).</figcaption>
+</figure>
+
+{% Aside %}
+Unlike a third-party picker widget, a browser picker is familiar to the user,
+has great platform-specific support, and is always maintained as part of the
+browser.
+{% endAside %}
+
+## How to show a picker {: #how-to }
+
+Calling `showPicker()` on an `<input>` element returns a promise that resolves
+when a browser picker is shown to the user. It must be called in response to a
+user gesture such as a touch gesture or mouse click; otherwise it will reject.
+For security reasons,it will also reject when it's called in a cross-origin
+iframe.
```

Comment:
It doesn't return a promise. It returns undefined, or throws an exception.

Reply:
😮‍💨  I'm the one who implemented in Chrome and I succeeded to write something wrong about it.
Thanks for catching this @domenic!  

![image](https://user-images.githubusercontent.com/634478/149472102-15b1ccd8-8e15-4f79-af35-23187076219a.png)